File: Python/string-1/non_start.py
'''
CodingBat Python Activity "non_start" from string-1.
codingbat.com
'''

failures = 0

def non_start(a, b):
    '''
Given 2 strings, return their concatenation, except omit the first char of each.
The strings will be at least length 1.


non_start('Hello', 'There') → 'ellohere'
non_start('java', 'code') → 'avaode'
non_start('shotl', 'java') → 'hotlava'
'''
    pass #Replace pass with your code to implement this function





def catchFailures(testCase, result):
    global failures
    
    try:
        assert eval(testCase) == result, "Test Case Failed: "+testCase+ " → "+str(result)
    except Exception as e:
        failures += 1
        print(e)
            


def test():
    global failures
    
    catchFailures("non_start('Hello', 'There')", 'ellohere')
    catchFailures("non_start('java', 'code')", 'avaode')
    catchFailures("non_start('shotl', 'java')", 'hotlava')
    catchFailures("non_start('ab', 'xy')", 'by')
    catchFailures("non_start('ab', 'x')", 'b')
    catchFailures("non_start('x', 'ac')", 'c')
    catchFailures("non_start('a', 'x')", '')
    catchFailures("non_start('kit', 'kat')", 'itat')
    catchFailures("non_start('mart', 'dart')", 'artart')

        
    if failures > 0:
        print("\r\n "+ str(failures) + " failed test case(s)   :(")
    else:
        print("Good job -- problem solved!")


test()

File: Java/String1/NonStart.java
/*
 * CodingBat Java Activity "nonStart" from String-1 section.
 * codingbat.com
 */

public class NonStart 
{
	private static int failures = 0;
	private static final String METHOD_NAME = "nonStart";
	
	/*
	 * INSTRUCTIONS
	 * Given two strings, return their concatenation, except omit the first char of each.
	 * The strings will be at least length 1.
	 * 
	 * nonStart("Hello", "There") -> "ellohere"
	 * nonStart("java", "code") -> "avaode"
	 * nonStart("shotl", "java") -> "hotlava"
	 */
	public static String nonStart(String a, String b)
	{
		//put your code ONLY here!
	}
	
	public static void main(String[] args)
	{
		testCase("\"Hello\"", "\"There\"", "\"ellohere\"");
		testCase("\"java\"", "\"code\"", "\"avaode\"");
		testCase("\"shotl\"", "\"java\"", "\"hotlava\"");
		testCase("\"ab\"", "\"xy\"", "\"by\"");
		testCase("\"ab\"", "\"x\"", "\"b\"");
		testCase("\"x\"", "\"ac\"", "\"c\"");
		testCase("\"a\"", "\"x\"", "\"\"");
		testCase("\"kit\"", "\"kat\"", "\"itat\"");
		testCase("\"mart\"", "\"dart\"", "\"artart\"");
		
		System.out.println();
		System.out.println(failures + " failed test cases.");
	}

	private static void testCase(String a, String b, String result)
	{
		String r;
		try
		{
			r = ""+ "\"" + nonStart(a.substring(1,a.length()-1), b.substring(1,b.length()-1)) + "\"";
			if (r.equals(""+result))
			{
				System.out.print("PASS: ");
			}
			else
			{
				System.out.println("FAIL: ");
				failures++;
			}
		}
		catch (Exception e)
		{
			r = e.getMessage();
			System.out.print("FAIL: ");
			failures++;
		}
		System.out.println(METHOD_NAME+"("+a+", "+b+") -> "+r + ", EXPECTED: "+result);
	}

}

